packages/kbn-text-based-editor/src/text_based_languages_editor.tsx

```diff
@@ -468,7 +468,9 @@ export const TextBasedLanguagesEditor = memo(function TextBasedLanguagesEditor({
               undefined,
               abortController
             ).result;
-            const columns = table?.columns.map((c) => ({ name: c.name, type: c.meta.type })) || [];
+            const columns =
+              // Forcing esType! here because esType will never be undefined
+              table?.columns.map((c) => ({ name: c.name, type: c.meta.esType! })) || [];
```

Comment:
Did you consider adjusting the type "lower" in the stack?

Reply:
I attempted to update DatatableColumnMeta and it caused 118 TS errors so I promptly reverted 😅

Reply:
:lol-rip:

In that case, maybe we should just default to something like `"unknown"` or `""` here. That makes me less nervous than the cast. The code should never hit that scenario, but if it does, at least it will be a string

Reply:
I added fallback to "unsupported" which I think is reasonable [`173255e` (#188600)](https://github.com/elastic/kibana/pull/188600/commits/173255e2134da1ef48cc165eb3f2206e29b92de6) since this case probably never happen

Reply:
I don't think we should filter the table columns though. If we do that, they will be out of sync with the rows. I think it's okay to have `unsupported` floating around the validator. Users would just get a message that the field is of the wrong type if they try to use it.